src/components/history/History.js

```diff
@@ -109,11 +119,19 @@ class History extends PureComponent {
       selectedMessageHasBeenClickedOnAgain ||
       isLoading
     ) {
-      this.needsInitialLoad = true
```

Comment:
While it looks like a great refactoring, but I'm bit worried it missed some cases. I suggest to leave this part as it is and refactor it separately in the future.